lib/Utils/Utilities.cpp

```diff
@@ -77,20 +114,22 @@ const llvm::Type *stripPointer(const llvm::Type *Pointer) {
   return Pointer;
 }
 
-bool isMangled(const string &Name) { return Name != llvm::demangle(Name); }
-
-vector<string> splitString(const string &Str, const string &Delimiter) {
-  vector<string> SplitStrings;
-  boost::split(SplitStrings, Str, boost::is_any_of(Delimiter),
-               boost::token_compress_on);
-  return SplitStrings;
-}
-
-ostream &operator<<(ostream &OS, const vector<bool> &Bits) {
-  for (auto Bit : Bits) {
-    OS << Bit;
+bool isMangled(llvm::StringRef Name) {
+  // See llvm/Demangle/Demangle.cpp
+  if (Name.startswith("_Z") || Name.startswith("___Z")) {
+    // Itanium ABI
+    return true;
+  }
+  if (Name.startswith("_R")) {
+    // Rust ABI
+    return true;
+  }
+  if (Name.startswith("_D")) {
+    // D ABI
+    return true;
   }
-  return OS;
+  // Microsoft ABI is a bit more complicated...
+  return Name != llvm::demangle(Name.str());
 }
 
 bool StringIDLess::operator()(const std::string &Lhs,
```

Comment:
please make a seperate PR for this change, this is unrelated to folder structure and too complex just to sneak it in

Reply:
Agreed, and tag me in the PR please I can add the stuff regarding the Swift ABI and mangling :)

Here we want to essentially check for "begins with '$s'".